src/main/java/mekanism/client/render/item/ChemicalFluidBarDecorator.java

```diff
@@ -0,0 +1,107 @@
+import com.mojang.blaze3d.vertex.PoseStack;
+import mekanism.api.chemical.ChemicalStack;
+import mekanism.api.chemical.IChemicalHandler;
+import mekanism.api.math.MathUtils;
+import mekanism.client.gui.GuiUtils;
+import mekanism.common.util.FluidUtils;
+import mekanism.common.util.StorageUtils;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.Font;
+import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.client.IItemDecorator;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.FluidUtil;
+import net.minecraftforge.fluids.capability.IFluidHandlerItem;
+
+import java.util.Optional;
+import java.util.function.Predicate;
+
+public class ChemicalFluidBarDecorator implements IItemDecorator {
+
+    private final Capability<? extends IChemicalHandler<?,?>>[] chemicalCaps;
+    private final boolean showFluid;
+    private final Predicate<ItemStack> visibleFor;
+
+    /**
+     * @param showFluid if the fluidcapability should be checked for display, display above chemicalCaps if both are present
+     * @param visibleFor checks if bars should be rendered for the given itemstack
+     * @param chemicalCaps the capabilities to be displayed in order, starting from the bottom
+     */
+    @SafeVarargs
+    public ChemicalFluidBarDecorator(boolean showFluid, Predicate<ItemStack> visibleFor, Capability<? extends IChemicalHandler<?, ?>>... chemicalCaps) {
+        this.showFluid = showFluid;
+        this.chemicalCaps = chemicalCaps;
+        this.visibleFor = visibleFor;
+    }
+
+    @Override
+    public boolean render(Font font, ItemStack stack, int xOffset, int yOffset, float blitOffset) {
+        if (!visibleFor.test(stack)) {
+            return false;
+        }
+        RenderSystem.disableDepthTest();
+        yOffset = yOffset + 12;
+        for (Capability<? extends IChemicalHandler<?,?>> chemicalCap : chemicalCaps) {
+            Optional<? extends IChemicalHandler<?, ?>> capabilityInstance = stack.getCapability(chemicalCap).resolve();
+            if (capabilityInstance.isPresent()) {
+                IChemicalHandler<?, ?> chemicalHandler = capabilityInstance.get();
+                if (chemicalHandler.getTanks() == 0) {
+                    continue;
+                }
+                int tank = getDisplayTank(chemicalHandler.getTanks());
+                if (tank != -1) {
+                    renderBarChemical(xOffset, yOffset, chemicalHandler.getChemicalInTank(tank), chemicalHandler.getTankCapacity(tank));
+                    yOffset--;
+                }
+            }
+        }
+
+        if (showFluid) {
+            Optional<IFluidHandlerItem> capabilityInstance = FluidUtil.getFluidHandler(stack).resolve();
+            if (capabilityInstance.isPresent()) {
+                IFluidHandlerItem fluidHandler = capabilityInstance.get();
+                if (fluidHandler.getTanks() != 0) {
```

Comment:
This isn't needed anymore because it is handled by the `tanks != -1` check below